frontend/NotificationViewer.vue

```diff
@@ -0,0 +1,294 @@
+                v-model="notification.state"
+                :value="'SENT_STATE'"
+                type="radio"
+                data-toggle="tooltip"
+                data-placement="top"
+                :title="notification.state === 'SENT_STATE' ? $t('mark_as_read') : $t('mark_as_unread')"
+                @click="markAsRead(notification)"
+            >
+            <span class="label label-primary">{{ notification.sent_at }}</span>
+            <!-- Disable the vue/no-v-html warning -->
+            <!-- eslint-disable-next-line -->
+            <span class="notification-text" :class="{ 'font-bold': notification.state === 'SENT_STATE' }" v-html="notification.payload" />
+          </div>
+        </a>
+      </li>
+    </ul>
+    <div
+        v-if="error"
+        class="alert alert-warning"
+        role="alert"
+    >
+      {{ error }}
+    </div>
+  </li>
+</template>
+
+<script>
+
+export default {
+  name: 'NotificationViewer',
+  props: {
+    url: {
+      type: String,
+      default: '',
+    },
+    interval: {
+      type: Number,
+      default: 300,
+    },
+  },
+  data() {
+    return {
+      notifications: [],
+      error: '',
+    };
+  },
+  computed: {
+    notReadNotificationsCount: function () {
+      return this.notifications.filter(notification => notification.state === 'SENT_STATE').length;
+    },
+  },
+  async created() {
+    await this.fetchNotifications();
+    this.timer = setInterval(this.fetchNotifications, this.interval * 1000);
+  },
+  mounted() {
+    // This next line let the dropdown menu open after clicking inside it. See the bootstrap source code here:
+    // https://github.com/twbs/bootstrap/blob/0b9c4a4007c44201dce9a6cc1a38407005c26c86/js/dropdown.js#L160
+    jQuery(document).on('click.bs.dropdown.data-api', '.notification-dropdown', e => e.stopPropagation());
+    // activate the tooltips
+    jQuery(() => jQuery('[data-toggle="tooltip"]').tooltip());
+  },
+  methods: {
+    fetchNotifications: async function () {
+      try {
+        if (this.url) {
+          const response = await fetch(this.url);
+          let newNotifications = await response.json();
+          this.notifications = newNotifications.results;
+        }
+      } catch (error) {
+        this.error = this.$t('error_fetch_notifications').concat(' (', error.statusText, ')');
+      }
+    },
+    markAsRead: async function (notification) {
+      try {
+        const response = await fetch(this.url.concat(notification.uuid), {
+          method: 'PATCH',
+          headers: {'X-CSRFToken': this.getCookie('csrftoken')},
+        });
+        if (response.status === 200) {
+          const notificationIndex = this.notifications.indexOf(notification);
+          let newNotification = await response.json();
+          this.$set(this.notifications, notificationIndex, newNotification);
+          // Change the bootstrap input radio tooltip
+          jQuery('#'.concat(notification.uuid))
```

Comment:
Bon c'est un avis personnel, mais partout où tu utilise `concat`, je préfère largement
```suggestion
          jQuery(`#${notification.uuid}`)
```